galgebra/printer.py

```diff
@@ -90,66 +90,7 @@ def isinteractive():  #Is ipython running
 
 
 def ostr(obj, dict_mode=False, indent=True):
-    """
-    Recursively convert iterated object (list/tuple/dict/set) to string.
-    """
-    def ostr_rec(obj, dict_mode):
-        ostr_s = ""
-        if isinstance(obj, Matrix):
-            ostr_s += str(obj)
-        elif isinstance(obj, tuple):
-            if len(obj) == 0:
-                ostr_s += '(),'
-            else:
-                ostr_s += '('
-                for obj_i in obj:
-                    ostr_s += ostr_rec(obj_i, dict_mode)
-                ostr_s = ostr_s[:-1] + '),'
```

Comment:
This `[:-1]` (and the others) was the cause of the matrix printing bug. This code seems to assume that `ostr_rec` always returns a string with a trailing comma, but line 99 does not, so this strips the `)` instead of the `,`.

The bug is irrelevant now the code doesn't exist.